[Docs] Add AVR and ARM examples to GPIO Commands (#6942)
qmk/qmk_firmware: docs/internals_gpio_control.md

The following functions can provide basic control of GPIOs and are found in `quantum/quantum.h`.

|Function              |Description                                                       | Old AVR Examples                               | Old ChibiOS/ARM Examples                      |
|----------------------|------------------------------------------------------------------|------------------------------------------------|-----------------------------------------------|

> |`setPinInput(pin)`    |Set pin as input with high impedance (High-Z)                     | DDRB &= ~(1<<2)                                | palSetLineMode(pin, PAL_MODE_INPUT)           |

Reviewer: Needs code tags, the pipe character messes up the table formatting :)

Author: lol, whoops, will fix.

The following functions can provide basic control of GPIOs and are found in `quantum/quantum.h`.

|Function              |Description                                                       | Old AVR Examples                               | Old ChibiOS/ARM Examples                        |
|----------------------|------------------------------------------------------------------|------------------------------------------------|-------------------------------------------------|
|`setPinInput(pin)`    |Set pin as input with high impedance (High-Z)                     | `DDRB &= ~(1<<2)`                              | `palSetLineMode(pin, PAL_MODE_INPUT)`           |

> |`setPinInputHigh(pin)`|Set pin as input with build in pull-up                            | `DDRB &= ~(1<<2); PORTB \|= (1<<2)`             | `palSetLineMode(pin, PAL_MODE_INPUT_PULLUP)`    |

Reviewer: ```suggestion
|`setPinInputHigh(pin)`|Set pin as input with builtin pull-up resistor                    | `DDRB &= ~(1<<2); PORTB \|= (1<<2)`             | `palSetLineMode(pin, PAL_MODE_INPUT_PULLUP)`    |
```

> |`setPinInputLow(pin)` |Set pin as input with build in pull-down (Supported only on STM32)| N/A (Not supported on AVR)                     | `palSetLineMode(pin, PAL_MODE_INPUT_PULLDOWN)`  |

Reviewer: ```suggestion
|`setPinInputLow(pin)` |Set pin as input with builtin pull-down resistor                  | N/A (Not supported on AVR)                     | `palSetLineMode(pin, PAL_MODE_INPUT_PULLDOWN)`  |
```

|`setPinOutput(pin)`   |Set pin as output                                                 | `DDRB \|= (1<<2)`                               | `palSetLineMode(pin, PAL_MODE_OUTPUT_PUSHPULL)` |
|`writePinHigh(pin)`   |Set pin level as high, assuming it is an output                   | `PORTB \|= (1<<2)`                              | `palSetLine(pin)`                               |
|`writePinLow(pin)`    |Set pin level as low, assuming it is an output                    | `PORTB &= ~(1<<2)`                             | `palClearLine(pin)`                             |
|`writePin(pin, level)`|Set pin level, assuming it is an output                           | `(level) ? PORTB \|= (1<<2) : PORTB &= ~(1<<2)` | `(level) ? palSetLine(pin) : palClearLine(pin)` |
|`readPin(pin)`        |Returns the level of the pin                                      | `_SFR_IO8(pin >> 4) & _BV(pin & 0xF)`          | `palReadLine(pin)`                              |

> ## Advanced Settings

Reviewer 2: Need a line break before this heading.

![image](https://user-images.githubusercontent.com/18669334/66342619-b2c36d80-e8fe-11e9-8435-ddf72ea232b9.png)

```suggestion

## Advanced Settings
```
